apps/sms/services/js/moz_mobile_message/moz_mobile_message_shim.js

```diff
@@ -71,6 +71,22 @@ function cloneThread(thread) {
   };
 }
 
+function clone(objectToBeCloned) {
+  // Basis checking.
+  if (!(objectToBeCloned instanceof Object)) {
+    return objectToBeCloned;
+  }
+
+  var objectClone = {};
+
+  // Clone each property.
+  for (var prop in objectToBeCloned) {
```

Comment:
Well, these properties are actually Inherited enumerable, so you can't list it even with `Object.getOwnPropertyNames`. The only way is using `for...in` in this case. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties for more details.